packages/frontend/component/src/components/attachment-viewer/utils.ts

```diff
@@ -0,0 +1,40 @@
+import { fileTypeFromBuffer } from 'file-type';
+
+export async function saveBufferToFile(url: string, filename: string) {
+  // given input url may not have correct mime type
+  const blob = await attachmentUrlToBlob(url);
+  if (!blob) {
+    return;
+  }
+
+  const blobUrl = URL.createObjectURL(blob);
+  const a = document.createElement('a');
+  a.href = blobUrl;
+  a.download = filename;
+  document.body.append(a);
+  a.click();
+  a.remove();
+  URL.revokeObjectURL(blobUrl);
+}
+
+export async function attachmentUrlToBlob(
+  url: string
+): Promise<Blob | undefined> {
+  const buffer = await fetch(url).then(response => response.arrayBuffer());
+
+  if (!buffer) {
+    console.warn('Could not get blob');
+    return;
+  }
+  try {
+    const type = await fileTypeFromBuffer(buffer);
```

Comment:
fileType is a very time and memory consuming operation. If we know that this is a pdf, we should directly set it to application/pdf